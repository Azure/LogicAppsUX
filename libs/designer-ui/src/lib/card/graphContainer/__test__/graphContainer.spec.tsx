import { describe, it, expect } from 'vitest';
import renderer from 'react-test-renderer';
import { GraphContainer } from '../index';

describe('GraphContainer component', () => {
  it('renders with default props', () => {
    const container = renderer.create(<GraphContainer id="test-container" />).toJSON();
    expect(container).toMatchSnapshot();
    expect(container).toHaveProperty('props.className');
    // Class names are now dynamic with makeStyles, we can't check for specific strings
    expect(container).toHaveProperty('props.data-automation-id', 'msla-graph-container-test_container');
  });

  it('renders with selected prop', () => {
    const container = renderer.create(<GraphContainer id="test-container" selected={true} />).toJSON();
    expect(container).toMatchSnapshot();
    expect(container).toHaveProperty('props.className');
    // With makeStyles, we can't check for specific class strings, but we verify the className exists
    expect(container).toHaveProperty('props.data-automation-id', 'msla-graph-container-test_container');
  });

  it('renders with active prop set to false', () => {
    const container = renderer.create(<GraphContainer id="test-container" active={false} />).toJSON();
    expect(container).toMatchSnapshot();
    expect(container).toHaveProperty('props.className');
    // With makeStyles, we can't check for specific class strings
    expect(container).toHaveProperty('props.data-automation-id', 'msla-graph-container-test_container');
  });

  it('renders with selected and active props', () => {
    const container = renderer.create(<GraphContainer id="test-container" selected={true} active={false} />).toJSON();
    expect(container).toMatchSnapshot();
    expect(container).toHaveProperty('props.className');
    // With makeStyles, we can't check for specific class strings
    expect(container).toHaveProperty('props.data-automation-id', 'msla-graph-container-test_container');
  });
});
