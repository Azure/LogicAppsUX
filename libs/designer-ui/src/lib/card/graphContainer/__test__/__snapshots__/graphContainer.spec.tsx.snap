// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GraphContainer component > renders with active prop set to false 1`] = `
<div
  className="___12cglbp_5g59060 fly5x3f f1l02sjl f10pi13n f1dmdbja f1ewtqcl f1erghxr f1ehz9de f1spoy8 fmb70yw ftac7j7 f1w0yd9v fdwcyh7 f1h0xgbp fzkkow9 fcdblym fg706s2 fjik90z fj3muxo f1akhkt f1aperda f1lxtadh"
  data-automation-id="msla-graph-container-test_container"
/>
`;

exports[`GraphContainer component > renders with default props 1`] = `
<div
  className="___12cglbp_5g59060 fly5x3f f1l02sjl f10pi13n f1dmdbja f1ewtqcl f1erghxr f1ehz9de f1spoy8 fmb70yw ftac7j7 f1w0yd9v fdwcyh7 f1h0xgbp fzkkow9 fcdblym fg706s2 fjik90z fj3muxo f1akhkt f1aperda f1lxtadh"
  data-automation-id="msla-graph-container-test_container"
/>
`;

exports[`GraphContainer component > renders with selected and active props 1`] = `
<div
  className="___1unu5ne_1fvunjk fly5x3f f1l02sjl f10pi13n f1dmdbja f1ewtqcl f1erghxr f1ehz9de f1spoy8 fmb70yw ftac7j7 f1w0yd9v fdwcyh7 f1h0xgbp fzkkow9 fcdblym fg706s2 fjik90z fc56jbi f1o6oagp fxmc7hf f1ml3t7l"
  data-automation-id="msla-graph-container-test_container"
/>
`;

exports[`GraphContainer component > renders with selected prop 1`] = `
<div
  className="___1unu5ne_1fvunjk fly5x3f f1l02sjl f10pi13n f1dmdbja f1ewtqcl f1erghxr f1ehz9de f1spoy8 fmb70yw ftac7j7 f1w0yd9v fdwcyh7 f1h0xgbp fzkkow9 fcdblym fg706s2 fjik90z fc56jbi f1o6oagp fxmc7hf f1ml3t7l"
  data-automation-id="msla-graph-container-test_container"
/>
`;
